trino/trino.go

```diff
@@ -1715,6 +1894,272 @@ func (qr *driverRows) fetch() error {
 	}
 }
 
+func (st *driverStmt) startSpoolingProtocolWorkers(ctx context.Context) {
+	st.usingSpooledProtocol = true
+
+	if st.spoolingWorkerCount == 0 {
+		st.spoolingWorkerCount = defaultSpoolingDownloadWorkers
+	}
+
+	if st.spoolingMaxOutOfOrderSegments == 0 {
+		st.spoolingMaxOutOfOrderSegments = defaultallowedOutOfOrder
+	}
+
+	downloadSegmentsCtx, cancelDownloadWorkers := context.WithCancel(context.WithoutCancel(ctx))
+	st.cancelDownloadWorkers = cancelDownloadWorkers
+	decodeSegmentCtx, cancelDecodersWorkers := context.WithCancel(context.WithoutCancel(ctx))
+	st.cancelDecodersWorkers = cancelDecodersWorkers
+
+	st.segmentsToProccess = make(chan segmentToProccess, 1000)
+	st.spooledSegmentsMetadata = make(chan spooledMetadata, st.spoolingMaxOutOfOrderSegments)
+	st.spooledSegmentToDecode = make(chan segmentToDecode, st.spoolingMaxOutOfOrderSegments)
+	st.spoolingRowsChannel = make(chan []queryData)
+	st.spoolingProcesserDone = make(chan struct{})
+	st.rateLimitDoneCh = make(chan struct{})
```

Comment:
This name is not accurate; it's used to wait for the segment dispatcher to finish.
```suggestion
	st.segmentDispatcherDoneCh = make(chan struct{})
```